tasks/yakimov_i_max_values_in_matrix_rows/tests/functional/functional.cpp

```diff
@@ -0,0 +1,57 @@
+#include <gtest/gtest.h>
+
+#include <array>
+#include <cstddef>
+#include <string>
+#include <tuple>
+
+#include "util/include/func_test_util.hpp"
+#include "util/include/util.hpp"
+#include "yakimov_i_max_values_in_matrix_rows/common/include/common.hpp"
+#include "yakimov_i_max_values_in_matrix_rows/mpi/include/ops_mpi.hpp"
+#include "yakimov_i_max_values_in_matrix_rows/seq/include/ops_seq.hpp"
+
+namespace yakimov_i_max_values_in_matrix_rows {
+
+class YakimovIMaxValuesInMatrixRowsFuncTests : public ppc::util::BaseRunFuncTests<InType, OutType, TestType> {
+ public:
+  static std::string PrintTestParam(const TestType &test_param) {
+    return std::to_string(std::get<0>(test_param)) + "_" + std::get<1>(test_param);
+  }
+
+ protected:
+  bool CheckTestOutputData(OutType &output_data) final {
+    return output_data > 0;
+  }
+
+  InType GetTestInputData() final {
+    TestType params = std::get<static_cast<std::size_t>(ppc::util::GTestParamIndex::kTestParams)>(GetParam());
+    return std::get<0>(params);
+  }
+};
+
+namespace {
+
+TEST_P(YakimovIMaxValuesInMatrixRowsFuncTests, MaxValuesInRows) {
+  ExecuteTest(GetParam());
+}
+
+const std::array<TestType, 5> kTestParam = {std::make_tuple(1, "tiny"), std::make_tuple(2, "very_small"),
```

Comment:
Please, add more tests. You don't even use all files you have uploaded. Also consider checking corner cases